internal/builders/golang/build.go

```diff
@@ -366,3 +368,32 @@ func hasMain(file *ast.File) bool {
 	}
 	return false
 }
+
+func addHeaderArtifactIfLibrary(ctx *context.Context, build config.Build, options api.Options) {
+	for _, s := range build.BuildDetails.Flags {
+		if s == "-buildmode=c-shared" || s == "-buildmode=c-archive" {
```

Comment:
one thing to be aware is that flags are templateable... so this might be an issue here...

Reply:
Fair point. I didn't think of that... How does this differ from the build pipe where we set the `so`, `dll`, etc extensions?
I guess the templates are resolved somewhere between here and then?
The only ideas I can think of right now would be to either:
* introduce a build-level setting to configure the buildmode
* have an archive setting for additional files, based on the binary output path

I guess you may have a better solution though!

Reply:
> Fair point. I didn't think of that... How does this differ from the build pipe where we set the so, dll, etc extensions?

It seems that it is also wrong 🤔 

On a second thought, I think that's not something people will template, so it might be fine. I'll add an issue to refactor that at some point though.